### ISSUE TITLE
Zenodo

### ISSUE THREAD

**Issue by 00sapo** [dlesbre/bibtex-autocomplete#7]
Some conferences in my field are used to publish papers to zenodo. Therefore, I would like to implement a lookup for zenodo. I've given a look at the code, and it should be similar to the DBLP API class, but I cannot completely get what are the supposed input and output types of each method.

Documentation for the Zenodo API is here: https://developers.zenodo.org/#list36

**Comment by dlesbre** [comment]
Yes it's non obvious because the inheritance structure is quite large. It inherits from JSON_AT_Lookup will perform queries using title (and author, if present) info, and expect a JSON response. If you copy the DBLP class you should only have to change the following:

-o verride `name` to a unique name, `path` and `domain` to the API's path and domain
- override `get_params` to return the query parameters (so `{'q': my_search_string, 'token':ACCESS_TOKEN}`)
- override `get_value` to parse your JSON into a BibtexEntry. 
  - The SafeJSON class I use is simply a class that returns None instead of failing if keys/indexes/whatever are invalid
  - [BibtexEntry](https://github.com/dlesbre/bibtex-autocomplete/blob/4aaf9bb12fbd2e5f41676655319c9105314e864c/bibtexautocomplete/bibtex/entry.py#L63) just has all standard bibtex field as attributes, with some hidden normalization logic. Most fields are string options. The author field is a list of a custom Author class.
- override FieldNames to the set of field names you define in `get_value`. This allows to skip queries for entries where all those fields are defined, since there is no data to add

Let me know if you have more specific questions

**Comment by dlesbre** [comment]
Since I haven't heard anymore from you I'll assume you figured it out. If not feel free to send me any further questions you might have!

**Comment by 00sapo** [comment]
I haven't actually been working on this. I'll come back on it when I will need it again...
